File: ARC100_199/ARC104/b.py
def main():
    N, S = input().split()
    N = int(N)
    ans = 0
    for i in range(N):
        A, T, C, G = 0, 0, 0, 0
        for j in range(i, N):
            if(S[j] == 'A'): A += 1
            if(S[j] == 'T'): T += 1
            if(S[j] == 'C'): C += 1
            if(S[j] == 'G'): G += 1
            if(A == T and C == G): ans += 1
    print(ans)


if __name__ == "__main__":
    main()

File: ARC100_199/ARC104/b.cpp
#include <iostream>
using namespace std;

int main() {
  int N;
  string S, str;
  cin >> N >> S;

  long long ans = 0;
  for (int i = 0; i < N; i++) {
    str = "";
    int A = 0, T = 0, C = 0, G = 0;
    for (int j = i; j < N; j++) {
      if (S[j] == 'A') A += 1;
      if (S[j] == 'T') T += 1;
      if (S[j] == 'C') C += 1;
      if (S[j] == 'G') G += 1;
      str += S[j];
      if(A == T && C == G) ans += 1;
    }
  }
  cout << ans << endl;
}